ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/hierarchies/OntologyHierarchiesPresenterImpl.java

```diff
@@ -0,0 +1,333 @@
+                                                 @Override
+                                                 public void onFailure(Throwable caught) {
+                                                     ErrorHandler.post(caught);
+                                                     tree.unmask();
+                                                 }
+
+                                                 @Override
+                                                 public void onSuccess(OntologyHierarchy result) {
+                                                     if (result != null) {
+                                                         ontologyUtil.addUnclassifiedChild(result);
+                                                         //Set the key for the current root (which won't appear in the tree, but will be the name of the tab)
+                                                         // which will allow the children to know the full path from its parent to node
+                                                         ontologyUtil.treeStoreModelKeyProvider(result);
+                                                         addHierarchies(tree.getStore(),
+                                                                        null,
+                                                                        result.getSubclasses());
+                                                         tree.unmask();
+                                                     }
+                                                 }
+                                             });
+    }
+
+    void addHierarchies(TreeStore<OntologyHierarchy> treeStore, OntologyHierarchy parent, List<OntologyHierarchy> children) {
+        if ((children == null)
+            || children.isEmpty()) {
+            return;
+        }
+        if (parent == null) {
+            treeStore.add(children);
+
+        } else {
+            treeStore.add(parent, children);
+        }
+
+        helperMap(children);
+
+        for (OntologyHierarchy hierarchy : children) {
+            addHierarchies(treeStore, hierarchy, hierarchy.getSubclasses());
+        }
+    }
+
+    void helperMap(List<OntologyHierarchy> children) {
+        for (OntologyHierarchy hierarchy : children) {
+            String iri = hierarchy.getIri();
+            List<OntologyHierarchy> hierarchies = iriToHierarchyMap.get(iri);
+            if (hierarchies == null) {
+                hierarchies = Lists.newArrayList();
+            }
+            hierarchies.add(hierarchy);
+            iriToHierarchyMap.put(hierarchy.getIri(), hierarchies);
+        }
+    }
+
+    @Override
+    public void onOntologyHierarchySelectionChanged(OntologyHierarchySelectionChangedEvent event) {
+        fireEvent(event);
+    }
+
+    @Override
+    public void onAppRatingDeselected(AppRatingDeselected event) {
+        final App appToUnRate = event.getApp();
```

Comment:
Just to note: a good chunk of the methods here were removed from `AppCategoriesPresenter` and placed here instead.